libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -2011,7 +2434,47 @@ const char * IotMqtt_OperationType( IotMqttOperationType_t operation )
 }
 
 /*-----------------------------------------------------------*/
+/* Convert the MQTT Status to IOT MQTT Status Code */
+IotMqttError_t convertReturnCode( MQTTStatus_t mqttStatus )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
 
+    switch( mqttStatus )
+    {
+        case MQTTSuccess:
+            status = IOT_MQTT_SUCCESS;
+            break;
+
+        case MQTTBadParameter:
+            status = IOT_MQTT_BAD_PARAMETER;
+            break;
+
+        case MQTTNoMemory:
+            status = IOT_MQTT_NO_MEMORY;
+            break;
+
+        case MQTTSendFailed:
+        case MQTTRecvFailed:
+            status = IOT_MQTT_NETWORK_ERROR;
+            break;
+
+        case MQTTBadResponse:
+            status = IOT_MQTT_BAD_RESPONSE;
+            break;
+
+        case MQTTServerRefused:
+            status = IOT_MQTT_SERVER_REFUSED;
+            break;
+
+        default:
```

Comment:
I think there are other error codes in the new library. It needs to be handled. If not handled those errors will be considered as success.